authentik/core/api/applications.py

```diff
@@ -147,6 +152,15 @@ def _get_allowed_applications(
                 applications.append(application)
         return applications
 
+    def _filter_applications_with_launch_url(
+        self, pagined_apps: Iterator[Application]
+    ) -> list[Application]:
+        applications = []
+        for app in pagined_apps:
+            if app.get_launch_url():
```

Comment:
It doesn't, because we change `Application.get_provider` to required the queryset to have been done with `Application.objects.with_provider()`, which avoids that double query.